src/main/java/com/app/rest/dao/UserDAO.java

```diff
@@ -0,0 +1,94 @@
+package com.app.rest.dao;
+
+import com.app.rest.exception.UserNotFoundException;
+import com.app.rest.exception.UserPersistenceException;
+import com.app.rest.model.dto.UserDTO;
+import com.app.rest.model.persistence.UserDetail;
+import com.app.rest.repository.UserRepo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Component;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class UserDAO {
+
+    @Autowired
+    private UserRepo userRepo;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserDAO.class);
+
+
+    public UserDTO retrieveUserById(Long id) throws UserNotFoundException, IllegalStateException {
```

Comment:
```suggestion
    public UserDTO retrieveUserById(Long id) throws UserNotFoundException, **IllegalStateException** {
```

Mejor usar una excepcion personalizada q una enmascarable como IllegalState, podria no ser catcheada afuera ya q no lo obliga